packages/plugin-git/sources/gitUtils.ts

```diff
@@ -0,0 +1,32 @@
+import {execUtils}                                    from '@berry/core';
+import {xfs, NodeFS, PortablePath, ppath, Filename}   from '@berry/fslib';
+
+const tmp = require('tmp');
+
+export async function clone(repo: string) {
+  const directory = await createTemporaryFolder()
+  const {code, stdout} = await execUtils.execvp(`git`, [`clone`, `${repo}`, `${directory}`], {cwd: directory});
+  // TODO: Support git clone errors
+  return directory;
+}
+
+function createTemporaryFolder(name?: Filename): Promise<PortablePath> {
+  return new Promise<PortablePath>((resolve, reject) => {
+    tmp.dir({unsafeCleanup: true}, (error: any, dirPath: any) => {
+      if (error) {
+        reject(error);
+      } else {
+        resolve(dirPath);
+      }
+    });
+  }).then(async dirPath => {
+    dirPath = await xfs.realpathPromise(NodeFS.toPortablePath(dirPath));
+
+    if (name) {
```

Comment:
You don't need this part anymore, right?

Reply:
That's right 👍 I was thinking that maybe we should move a temp dir creation to some utility, like `fs.ts`? (we have such a file in test package only) 🤔

Reply:
This https://github.com/yarnpkg/berry/pull/325 introduced utils in `xfs` for tmp directory creation: `mktempSync` and `mktempPromise`. I will use one of these instead our custom implementation 👍